### PR TITLE
Enable asynchronous message handling.

### PR DESCRIPTION
**Pull request by xiaodong-izettle**
Ping @iZettle/transactions, thanks.

### CODE REVIEW

File: izettle-messaging/src/main/java/com/izettle/messaging/QueueProcessor.java [hunk 1]
```diff
@@ -27,54 +28,101 @@
     private final String name;
 
     private static final int MAXIMUM_NUMBER_OF_MESSAGES_TO_RECEIVE = 10;
-    private static final int MESSAGE_WAIT_SECONDS = 20;
+    private static final Integer MESSAGE_WAIT_SECONDS = Integer.valueOf(20);
```
Comment (nzroller) [inline izettle-messaging/src/main/java/com/izettle/messaging/QueueProcessor.java]: `private static final Integer MESSAGE_WAIT_SECONDS = 20;` ?

File: izettle-messaging/src/main/java/com/izettle/messaging/QueueProcessor.java [hunk 2]
```diff
@@ -107,14 +155,18 @@ private void pollMessageQueue(String messageQueueUrl, boolean useLongPolling) th
...
-            LOG.info("Client abort.");
+            LOG.info("Client abort receive message.");
             messages = Collections.emptyList();
         } catch (AmazonClientException e) {
             throw new MessagingException("Failed to poll message queue.", e);
         }
-
-        if (!empty(messages)) {
-            handleMessages(messages, messageQueueUrl);
+        List<Message> messagesToHandle = messages;
```
Comment (nzroller) [inline izettle-messaging/src/main/java/com/izettle/messaging/QueueProcessor.java]: Why do we need this variable and can't use messages directly?

Comment (xiaodong-izettle) [inline izettle-messaging/src/main/java/com/izettle/messaging/QueueProcessor.java]: Thanks for the discussion. :-)

File: izettle-messaging/src/main/java/com/izettle/messaging/QueueProcessor.java (hunk 1 above)
Comment (xiaodong-izettle) [inline izettle-messaging/src/main/java/com/izettle/messaging/QueueProcessor.java]: Well, actually, under the cover, they are the same. :-)

But I'll change. :)

**Comment by xiaodong-izettle** [comment]
OK, I'm done here, @nzroller, could you please have a look? Thanks. :)

**Comment by nzroller** [comment]
:+1: nice work @xiaodong-izettle

**Comment by xiaodong-izettle** [comment]
Thanks, @nzroller. :)

**Pull request merged**